### PR TITLE
Competition requirements added as per sprint

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 1]
```diff
@@ -37,7 +37,16 @@ public IEnumerable<Competition> GetCompetitions()
         public IEnumerable<Competition> GetAllCompetitions()
        {
             //get the competitions where(Status == true / false)
-             return _context.Competitions.ToArray<Competition>().OrderBy(x=> (x.Status == false)).ThenBy(x =>(x.Name));
+            List<Competition> CompList = (from C in _context.Competitions
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: try naming variables in fill names

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 2]
```diff
@@ -37,7 +37,16 @@ public IEnumerable<Competition> GetCompetitions()
...
+                                         select C).OrderBy(c => c.Status == false).ThenBy(x => x.Name).ToList<Competition>();
+            //get the list of Requirements for Each Competition
+            foreach (Competition C in CompList){
+                List<Requirement> ReqList = (from R in _context.Requirements
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: ReqList is meaningless

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 3]
```diff
@@ -92,61 +94,59 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
...
-                        _context.SaveChangesAsync();
+        public async Task<IActionResult> UpdateCompetitionByRequirements(int compID, [FromBody] Competition Comp)
+        { 
+            if (ModelState.IsValid){
+                //error handling, check if client provided valid data
+                if (Comp == null){
+                    return new BadRequestObjectResult("competiton cannot be null");
+                }
+                else {
+                    Competition dbComp;//assume the competition does not exist
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: DbComp also a bit meaningless

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs [hunk 4]
```diff
@@ -92,61 +94,59 @@ public async Task<IActionResult> UpdateCompetition(int id, [FromBody]Competition
...
+                    using (_context) {
+                        dbComp = _context.Competitions.Where(C => C.Id == Comp.Id).FirstOrDefault<Competition>();
+                        if (dbComp == null) {
+                            return BadRequest("Cannot update competition, no such competition!");
+                        }
+                        foreach (Requirement R in Comp.RequirementsList){
+                            Requirement Temp = new Requirement {
```
Comment (kyle-manganyi) [inline SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: TempRequirement would have been better

File: SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs (hunk 1 above)
Comment (mthembbu) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: Like instead of CompList or CompetitionList?

Comment (kyle-manganyi) [reply SkietbaanBE/SkietbaanBE/Controllers/CompetitionController.cs]: yes

**Pull request merged**